src/topp/PercolatorAdapter.cpp

```diff
@@ -0,0 +1,1026 @@
+    registerIntOption_("seed", "<value>", 1, "Setting seed of the random number generator.", !is_required, is_advanced_option);
+    registerIntOption_("doc", "<value>", 0, "Include description of correct features", !is_required, is_advanced_option);
+    registerFlag_("klammer", "Retention time features calculated as in Klammer et al. Only available if -doc is set", is_advanced_option);
+    registerInputFile_("fasta", "<file>", "", "Provide the fasta file as the argument to this flag, which will be used for protein grouping based on an in-silico digest (only valid if option -protein-level-fdrs is active).", !is_required, is_advanced_option);
+    setValidFormats_("fasta", ListUtils::create<String>("FASTA"));
+    registerStringOption_("decoy-pattern", "<value>", "random", "Define the text pattern to identify the decoy proteins and/or PSMs, set this up if the label that identifies the decoys in the database is not the default (Only valid if option -protein-level-fdrs is active).", !is_required, is_advanced_option);
+    registerFlag_("post-processing-tdc", "Use target-decoy competition to assign q-values and PEPs.", is_advanced_option);
+  }
+  
+  // TODO replace with TopPerc::getScanMergeKey
+  String getScanIdentifier_(vector<PeptideIdentification>::iterator it, vector<PeptideIdentification>::iterator start)
+  {
+    // MSGF+ uses this field, is empty if not specified
+    String scan_identifier = it->getMetaValue("spectrum_reference");
+    if (scan_identifier.empty())
+    {
+      // XTandem uses this (integer) field
+      // these ids are 1-based in contrast to the index which is 0-based. This might be problematic to use for merging
+      if (it->metaValueExists("spectrum_id") && !it->getMetaValue("spectrum_id").toString().empty())
+      {
+        scan_identifier = "scan=" + it->getMetaValue("spectrum_id").toString();
+      }
+      else
+      {
+        scan_identifier = "index=" + String(it - start + 1);
+        LOG_WARN << "no known spectrum identifiers, using index [1,n] - use at own risk." << endl;
+      }
+    }
+    return scan_identifier.removeWhitespaces();
+  }
+  
+  // TODO replace with TopPerc::getScanMergeKey
+  Int getScanNumber_(String scan_identifier)
+  {
+    Int scan_number = 0;
+    StringList fields = ListUtils::create<String>(scan_identifier);
+    for (StringList::const_iterator it = fields.begin(); it != fields.end(); ++it)
+    {
+      // if scan number is not available, use the scan index
+      Size idx = 0;
+      if ((idx = it->find("scan=")) != string::npos) 
+      {
+        scan_number = it->substr(idx + 5).toInt();
+        break;
+      }
+      else if ((idx = it->find("index=")) != string::npos) 
+      {
+        scan_number = it->substr(idx + 6).toInt();
+      }
+    }
+    return scan_number;
+  }
+  
+  // Function adapted from Enzyme.h in Percolator converter
+  bool isEnz_(const char& n, const char& c, string& enz)
```

Comment:
I haven't heard back, but a look into the matter left me with the feeling that this is not easily accomplished. Will leave like is for now, we might want to open an issue about once merged.